plugins/shared-react/src/utils/pipeline-utils/pipeline-utils.test.ts

```diff
@@ -0,0 +1,36 @@
+import { ComputedStatus } from '../../types';
+import { getRunStatusColor } from './pipeline-utils';
+
+describe('getRunStatusColor should handle ComputedStatus values', () => {
+  it('should expect all but PipelineNotStarted to produce a non-default result', () => {
+    // Verify that we cover colour states for all the ComputedStatus values
+    const failCase = 'PipelineNotStarted';
+    const defaultCase = getRunStatusColor(ComputedStatus[failCase]);
+    const allOtherStatuses = Object.keys(ComputedStatus)
+      .filter(
+        status =>
+          status !== failCase &&
+          ComputedStatus[status as keyof typeof ComputedStatus] !==
+            ComputedStatus.Other,
+      )
+      .map(status => ComputedStatus[status as keyof typeof ComputedStatus]);
+
+    expect(allOtherStatuses).not.toHaveLength(0);
+    allOtherStatuses.forEach(statusValue => {
+      const { message } = getRunStatusColor(statusValue);
+
+      expect(defaultCase.message).not.toEqual(message);
+    });
+  });
+
+  it('should expect all status colors to return visible text to show as a descriptor of the color', () => {
+    const runStates = Object.values(ComputedStatus);
+
+    expect(runStates).not.toHaveLength(0);
+    runStates.forEach(statusValue => {
+      const { message } = getRunStatusColor(statusValue);
+
+      expect(message).not.toHaveLength(0);
```

Comment:
even in the default state `getRunStatusColor` returns a message so there will always be length